src/ExpressionEvaluator/Core/Test/ExpressionCompiler/InspectionContextFactory.cs

```diff
@@ -63,9 +62,36 @@ internal override string GetObjectTypeNameById(string id)
 
             private string GetType(string id)
             {
-                string type;
-                _types.TryGetValue(id, out type);
-                return type;
+                var alias = _variables.FirstOrDefault(a => a.FullName == id);
+                return alias.Type;
+            }
+
+            private static PseudoVariableKind GetPseudoVariableKind(string name)
+            {
+                if (name.StartsWith("$", StringComparison.Ordinal))
+                {
+                    if (name.Equals("$exception", StringComparison.InvariantCultureIgnoreCase))
+                    {
+                        return PseudoVariableKind.Exception;
+                    }
+                    else if (name.Equals("$stowedexception", StringComparison.InvariantCultureIgnoreCase))
+                    {
+                        return PseudoVariableKind.StowedException;
+                    }
+                    else if (name.StartsWith("$ReturnValue", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
StringComparison.Ordinal?  (or else make lower case?)  The fact that it's different than the others caught my eye.